app/controllers/time_slot_controller.rb

```diff
@@ -47,28 +47,19 @@ def cancel
   end
 
   def index
-    # FIXME We need to change this relation to has_one or enhance the logic here
-    @appointment = current_patient.appointments.last
+    @appointment = current_patient.current_appointment
     @ubs = @appointment.try(:ubs)
 
-    @gap_in_days = params[:gap_in_days].to_i || 0
+    @gap_in_days = slot_params[:gap_in_days].to_i || 0
     @current_day = Time.zone.now + @gap_in_days.days
 
-    @gap_in_days > TimeSlotController::SLOTS_WINDOW_IN_DAYS
-
-    @time_slots = []
-    if !@current_day.past? && !@current_day.sunday? && !(@gap_in_days > TimeSlotController::SLOTS_WINDOW_IN_DAYS)
-      @time_slots = Ubs.all.where(active: true).each_with_object({}) do |ubs, memo|
+    @time_slots = {}
```

Comment:
No need for initiating this variable I think. The result from `Ubs.where` should be an `Array` so this will be overwritten there.
```suggestion
```

Reply:
Then again, if at some point we remove this `@time_slots.present?` It would break on the `@time_slots.each` afterwards. So `[]` would be useful :sweat_smile: So one of two options lets initiate it as a `[]` or let's not initiate this variable. WDYT?

Reply:
Yeah, the idea was to maintain a fallback in case the `unless` block isn't ran. But `@time_slots` is a hash here (check `each_with_object`), so I believe the best fallback is a hash as well. Makes sense?

Reply:
You right, didn't see that. Let's keep it a hash them.